### PR TITLE
Jest unit test set-up

### LINKED ISSUE #105: Code main process
**Is this related to a problem? Please describe.**
We need to have a main process separate from phoneme factory and animation processor.

When the code runs, the user inputs are extracted and sent to the phoneme factory, then the output is returned, and then the program continues with the animation processor.

### LINKED ISSUE #117: Issue with npm start and npm build
**Describe the bug**

Issue running 'npm start'. The contents of the build folder are deleted during run time. 

Changes need to be made to 'npm build' 

More context in the discord chat in the 'bug report' section

### PR DESCRIPTION
**Pull request by edouarde1**
Everything, but unit tests haha 

A lot of stuff on this branch will help with error handling. Phoneme Factory is wrapped in a function, the output phoneme contents are handled as a json object. This object represents mouthCues as a list of mouthCue types. 

### Bug Fix (src/package.json)
- Earlier the `npm build` script would delete the build directory during runtime which would cause issues running the code 

### Main Process (index.ts) 
- Edited the main process (index.ts) file to include a main that can run user inputs into the phoneme factory and return a json output of the contents

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -482,3 +482,11 @@ fabric.properties
 
 # Rhubarb Ignore
 **/rhubarb
+
+# Ignore Build/Dist Dir
+speaking-portal-project/build
```
Comment (LoganParker) [inline .gitignore:487]: Odd that some files in the build folder are still being included, I wonder if the path could be off?

File: speaking-portal-project/build/index.d.ts [hunk 2]
```diff
@@ -1 +0,0 @@
-export {};
```
Comment (LoganParker) [inline speaking-portal-project/build/index.d.ts:1]: I think potentially this was pushed before the gitignore was updated causing it to be included - I wonder if we could remove this?

File: speaking-portal-project/build/index.js [hunk 3]
```diff
@@ -1,77 +0,0 @@
-"use strict";
```
Comment (LoganParker) [inline speaking-portal-project/build/index.js:1]: Same comment as above, I think we should remove these before merging

File: speaking-portal-project/build/index.js.map [hunk 4]
```diff
@@ -1 +0,0 @@
-{"version":3,"file":"index.js","sourceRoot":"","sources":["../src/index.tsx"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,6CAAiD;AACjD,wDAA4B;AAC5B,0CAAmB;AAGnB,IAAM,IAAI,GAAG,mBAAI,CAAC,SAAS,CAAC,OAAO,CAAC,oBAAoB,CAAC,CAAC,IAAI,CAAC,CAAA;AAE/D,SAAe,gBAAgB,CAAE,eAAwB,EAAE,cAAyB;;;;;;oBAEhF,OAAO,CAAC,GAAG,CAAC,+BAAwB,IAAA,kBAAG,GAAE,CAAE,CAAC,CAAA;oBAI5C,IAAI;wBACA,IAAA,oBAAK,EAAC,WAAW,CAAC,CAAA;wBAClB,OAAO,CAAC,GAAG,CAAC,yBAAkB,IAAA,kBAAG,GAAE,CAAE,CAAC,CAAA;qBACzC;oBAAC,OAAM,GAAG,EAAE;wBACT,OAAO,CAAC,GAAG,CAAC,yBAAkB,GAAG,CAAE,CAAC,CAAA;qBACvC;oBAID,OAAO,CAAC,GAAG,CAAC,sBAAsB,CAAC,CAAA;oBAClB,WAAO,IAAI,CAAC,8EAAqE,cAAc,kCAAwB,eAAe,CAAE,CAAC,EAAA;;oBAAnJ,MAAM,GAAI,CAAA,SAAyI,CAAA,OAA7I;oBACb,OAAO,CAAC,GAAG,CAAC,sBAAsB,CAAC,CAAA;oBAEnC,IAAI,MAAM,EAAE;wBACR,OAAO,CAAC,GAAG,CAAC,MAAM,CAAC,CAAA;qBACtB;oBAGD,OAAO,CAAC,GAAG,CAAC,8BAA8B,CAAC,CAAA;oBACrC,IAAI,GAAG,IAAI,CAAC,KAAK,CAAC,YAAE,CAAC,YAAY,CAAC,aAAa,EAAE,MAAM,CAAC,CAAC,CAAA;oBACzD,SAAS,GAAe,IAAI,CAAC,SAAuB,CAAA;oBAE1D,WAAO,SAAS,EAAA;;;;CACnB;AACE,IAAM,OAAO,GAAI,gBAAgB,CAAC,cAAc,EAAC,aAAa,CAAC,CAAA"}
```
Comment (LoganParker) [inline speaking-portal-project/build/index.js.map:1]: again same as above, 🚫build files🚫

File: speaking-portal-project/package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "scripts": {
     "start:dev": "nodemon",
     "start": "npm run build && node build/index.js",
-    "build": "rimraf build && tsc",
```
Comment (LoganParker) [inline speaking-portal-project/package.json:9]: Ill look further into this to see why it was causing an issue - we should remove rimraf from the dependencies if we decide its not necessary

**Review by LoganParker — CHANGES_REQUESTED** [review]
Looks good! We should make sure we remove those build files but otherwise all great :)

File: .gitignore (hunk 1 above)
Comment (edouarde1) [reply .gitignore:487]: I think it works now

**Review by vnajack — APPROVED** [review]
Thank you for re-organizing the rhubarb functions. I got it to work no problem on my machine!

**Review by LoganParker — APPROVED** [review]
This looks great!! Thank you for doing all of this, it makes a lot of sense now :)

**Pull request merged**